luavm/src/lib/mod.rs

```diff
@@ -0,0 +1,45 @@
+extern crate luacompiler;
+
+mod instructions;
+mod lua_value;
+
+use instructions::{arithmetic_operators::*, loads::*};
+use lua_value::{LuaNil, LuaValue};
+use luacompiler::bytecode::{instructions::opcode, LuaBytecode};
+use std::cell::RefCell;
+
+/// Represents a `LuaBytecode` interpreter.
+pub struct Vm {
+    pub bytecode: LuaBytecode,
+    pub registers: RefCell<Vec<Box<LuaValue>>>,
```

Comment:
Unless there's a good reason, we really don't want this to be a `RefCell`, as that is a very slow (and cumbersome to use) mechanism.

Reply:
I pass the vm to the [opcode handlers](https://github.com/rbartlensky/Lua-interpreter/pull/6/files/229c924eb01fe6b2fbc56214bb696db08971e466#diff-f0ae2c4a2151409996f150916404a6a5R41). Each [handler](https://github.com/rbartlensky/Lua-interpreter/pull/6/files/229c924eb01fe6b2fbc56214bb696db08971e466#diff-90a3a10505d23a91ce842a81a569eb84R5) needs to change the state of the registers. If I get rid of `RefCell`, this means that `eval` needs to take a mutable reference to self, but then I cannot pass self to the handlers anymore because `eval` has already mutably borrowed self.

Reply:
You're going to need to come up with another, probably simpler, design. We really can't afford the `RefCell` overhead here: this is one of the two most performance critical parts of the whole interpreter.